### ISSUE TITLE
Dense Layer Tracker

### ISSUE THREAD

**Issue by karanchahal** [karanchahal/buildTensorflow#20]
Now we can get to building the interesting Deep Learning layers ! now that our base of modular automatic differentiation is setup with some operations being performed on the GPu, we are set !

We've come a long way and let's appreciate that. But so much more to be done ! 

This issue tracks the development of the Dense Layer (Fully Conected layer)

**Comment by karanchahal** [comment]
SO for this feature I was thinking having a structure somewhat like this

```c++


template<typename T>
class Dense{
    Tensor<T>* weights; // size is input*output

    Tensor<T>* forward(Tensor<T> *x) {
        auto dot = tensorOps::dot(x,weights);
        auto *logits = tensorOps::sigmoid(dot);
        return logits;
    }

}

```

What do you think the constructor should have @uditarora ? I was thinking input size, output size to initlialise the weights. 

Also don't know how we should go on integrating dropout in this. 

Some code updates are :+1: 
1. Put matrix and tensor overloads in separate file under overloads/matrix.h and overloads/tensor.h
2. Write operation for sigmoid, with efficient backOp: dw = sigmoid(x)(1- sigmoid(x))
3. Use directory structure of layers/dense.h , layers/backNorm.h etc 


Lemme know what you think @uditarora :)

**Comment by karanchahal** [comment]
I've used the Glorot weight initialisation scheme which draws samples from a uniform distribution having variance (2/fan_in+fan_out) and mean 0.

Fan in and fan out mean the number of input neurons and the number of output neurons.

**Comment by karanchahal** [comment]
I've coded the Dense layer will raise a pull request for the same with tests soon. ALso I'll integrate a CI/CD system based on circle ci to run tests with each pull request.

**Comment by karanchahal** [comment]
Done, this issue is handled by #21

**Comment by karanchahal** [comment]
Fixed issue via #21
